pkg/apis/crd/v1alpha1/types.go

```diff
@@ -291,3 +291,157 @@ type TLSProtocol struct {
+	// only for live-traffic Traceflow.
+	IP string `json:"ip,omitempty"`
+}
+
+// Destination describes the destination spec of the traceflow.
+type Destination struct {
+	// Namespace is the destination namespace.
+	Namespace string `json:"namespace,omitempty"`
+	// Pod is the destination pod, exclusive with destination service.
+	Pod string `json:"pod,omitempty"`
+	// Service is the destination service, exclusive with destination pod.
+	Service string `json:"service,omitempty"`
+	// IP is the destination IPv4 or IPv6 address.
+	IP string `json:"ip,omitempty"`
+}
+
+// IPHeader describes spec of an IPv4 header.
+type IPHeader struct {
+	// Protocol is the IP protocol.
+	Protocol int32 `json:"protocol,omitempty" yaml:"protocol,omitempty"`
+	// TTL is the IP TTL.
+	TTL int32 `json:"ttl,omitempty" yaml:"ttl,omitempty"`
+	// Flags is the flags for IP.
+	Flags int32 `json:"flags,omitempty" yaml:"flags,omitempty"`
+}
+
+// IPv6Header describes spec of an IPv6 header.
+type IPv6Header struct {
+	// NextHeader is the IPv6 protocol.
+	NextHeader *int32 `json:"nextHeader,omitempty" yaml:"nextHeader,omitempty"`
+	// HopLimit is the IPv6 Hop Limit.
+	HopLimit int32 `json:"hopLimit,omitempty" yaml:"hopLimit,omitempty"`
+}
+
+// TransportHeader describes spec of a TransportHeader.
+type TransportHeader struct {
+	ICMP *ICMPEchoRequestHeader `json:"icmp,omitempty" yaml:"icmp,omitempty"`
+	UDP  *UDPHeader             `json:"udp,omitempty" yaml:"udp,omitempty"`
+	TCP  *TCPHeader             `json:"tcp,omitempty" yaml:"tcp,omitempty"`
+}
+
+// ICMPEchoRequestHeader describes spec of an ICMP echo request header.
+type ICMPEchoRequestHeader struct {
+	// ID is the ICMPEchoRequestHeader ID.
+	ID int32 `json:"id,omitempty"`
+	// Sequence is the ICMPEchoRequestHeader sequence.
+	Sequence int32 `json:"sequence,omitempty"`
+}
+
+// UDPHeader describes spec of a UDP header.
+type UDPHeader struct {
+	// SrcPort is the source port.
+	SrcPort int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort int32 `json:"dstPort,omitempty"`
+}
+
+// TCPHeader describes spec of a TCP header.
+type TCPHeader struct {
+	// SrcPort is the source port.
+	SrcPort int32 `json:"srcPort,omitempty"`
+	// DstPort is the destination port.
+	DstPort int32 `json:"dstPort,omitempty"`
+	// Flags are flags in the header.
+	Flags int32 `json:"flags,omitempty"`
+}
+
+// Packet includes header info.
+type Packet struct {
+	SrcIP string `json:"srcIP,omitempty"`
+	DstIP string `json:"dstIP,omitempty"`
+	// Length is the IP packet length (includes the IPv4 or IPv6 header length).
+	Length          int32           `json:"length,omitempty"`
+	IPHeader        IPHeader        `json:"ipHeader,omitempty"`
+	IPv6Header      *IPv6Header     `json:"ipv6Header,omitempty"`
+	TransportHeader TransportHeader `json:"transportHeader,omitempty"`
+}
+
+type PacketCaptureType string
+
+const (
+	FirstNCapture PacketCaptureType = "FirstNCapture"
+)
+
+type FirstNCaptureConfig struct {
```

Comment:
this should be renamed to `PacketCaptureFirstNConfig`

But I am a bit confused, `FirstN` means "first sequence of N packets", so why do we make it a separate struct? What kind of other config parameter do we expect in this struct in the future, besides `Number`?

Reply:
We want to support different cpauture methods in the future. For example, we may have a futre capture type called `IntervalCapture` which doing capture 1 packet 100ms. so there should a new config struct for that type too. So each type should have a dedicated config struct for them, no matter what's the num of paramaters.

OK, i see a problem here. It should be a struct like this,right?

```golang
type PacketCaptureSpec struct {
    CaptureType ...
    CaptureConfig CaptureConfig
}

type CaptureConfig  struct {
   FirstN *FirstN 
   ...
}
```

Reply:
updated. Now it looks likes 

```yaml
  captureConfig:
    firstN:
      number: 5
```
no extra `type` field needed.